lib/logidze/model.rb

```diff
@@ -117,6 +121,18 @@ def switch_to!(version)
       self.class.without_logging { save! }
     end
 
+    def association(name)
+      association = super
+
+      if logidze_past?
+        association.singleton_class.prepend Logidze::VersionedAssociation
+      end
+
+      association.reload if @logidze_requested_ts
```

Comment:
Hm, but we don't want to reload the association on every call. So we have to add another variable to track associations reload (btw, we should track each association separately) to avoid double-reload:

```ruby
post = Post.find(id)

# first load, without versions
post.comments

post.at!(2.days.ago)

# here we should reload (it would be better to reset, not reload, btw)
post.comments.first

# no reset/reload here
post.comments.second

# with fresh object
post = Post.find(id).at(2.days.ago)

# no reset/reload, 'cause association hasn't been loaded before
post.comments
```